cmd/retry-storage.go

```diff
@@ -189,36 +190,59 @@ func (f retryStorage) DeleteFile(volume, path string) (err error) {
 	return err
 }
 
-// Connect and attempt to load the format from a disconnected node.
-func (f retryStorage) reInit() (err error) {
-	err = f.remoteStorage.Close()
-	if err != nil {
-		return err
-	}
-	err = f.remoteStorage.Init()
-	if err == nil {
-		_, err = loadFormat(f.remoteStorage)
-		// For load format returning network shutdown
-		// we now treat it like disk not available.
-		if err == rpc.ErrShutdown {
-			err = errDiskNotFound
-		}
-		return err
-	}
-	if err == rpc.ErrShutdown {
-		err = errDiskNotFound
-	}
-	return err
-}
-
 // RenameFile - a retryable implementation of renaming a file.
 func (f retryStorage) RenameFile(srcVolume, srcPath, dstVolume, dstPath string) (err error) {
 	err = f.remoteStorage.RenameFile(srcVolume, srcPath, dstVolume, dstPath)
-	if err == rpc.ErrShutdown {
+	if err == rpc.ErrShutdown || err == errDiskNotFound {
 		err = f.reInit()
 		if err == nil {
 			return f.remoteStorage.RenameFile(srcVolume, srcPath, dstVolume, dstPath)
 		}
 	}
 	return err
 }
+
+// Connect and attempt to load the format from a disconnected node,
+// attempts three times before giving up.
+func (f retryStorage) reInit() (err error) {
+	err = f.remoteStorage.Close()
+	if err != nil {
```

Comment:
Should we exit on receiving an error here? It is likely that we might see the same network error again, especially if the underlying connection was closed in an earlier RPC call. I think we can ignore error on close in ``reInit``. Thoughts?